tests/pgautofailover_utils.py

```diff
@@ -814,17 +830,65 @@ def get_nodeid(self):
         self.state = json.loads(out)
         return self.state['state']['nodeId']
 
+    def jsDict(self, lsn="0/1", isPrimary=False):
+        """
+        Returns a python dict with the information to fill-in a JSON
+        representation of the node.
+        """
+        return {"node_id": self.get_nodeid(),
+                "node_name": self.name,
+                "node_host": str(self.vnode.address),
+                "node_port": self.port,
+                "node_lsn": lsn,
+                "node_is_primary": isPrimary}
+
     def get_local_state(self):
         """
         Fetch the assigned_state from the pg_autoctl state file.
         """
         command = PGAutoCtl(self)
-        out, err, ret = command.execute("get node id", 'do', 'fsm', 'state')
+        out, err, ret = command.execute("get node id", '-vv', 'do', 'fsm', 'state')
 
         self.state = json.loads(out)
         return (self.state['state']['current_role'],
                 self.state['state']['assigned_role'])
 
+    def wait_until_local_state(self, target_state,
```

Comment:
Good point. I even think this should be the default behaviour rather than an option, and I'm not sure about the need for a `--no-wait` option actually. Implemented that way now, with a timeout of 30s as usual.